File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        map<char,int>mp;
        for(int i=0;i<s.size();i++){
            mp[s[i]]++;
        }
        for(int j=0;j<t.size();j++){
            mp[t[j]]--;
        }
        for(auto it:mp){
            if(it.second !=0){
                return false;
            }
        }
        return true;
    }
};

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    # M-1:
    # def isAnagram(self, s: str, t: str) -> bool:
    #     return sorted(s) == sorted(t)

    # M-2
    #Counter(s) counts frequency of each character.
	#Compares both frequency maps.
    # def isAnagram(self, s: str, t: str) -> bool:
    #     return Counter(s) == Counter(t)
    
    # M-3
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        freq = {}

        # Count characters in s
        for ch in s:
            freq[ch] = freq.get(ch, 0) + 1

        # Subtract counts using t
        for ch in t:
            if ch not in freq or freq[ch] == 0:
                return False
            freq[ch] -= 1

        return True